### PR TITLE
[WAL-1391] Auth header sig

### PR DESCRIPTION
**Pull request by Ifropc**

### CODE REVIEW

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/AuthHeaderSigner.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  fun createBuilder(timeExp: Instant, claims: Map<String, String>): JwtBuilder {
+    return Jwts.builder().issuedAt(Date.from(Instant.now())).expiration(Date.from(timeExp)).also {
+      builder ->
+      claims.forEach { builder.claim(it.key, it.value) }
+    }
+  }
+}
+
+open class DomainAuthHeaderSigner(
```
Comment (CassioMG) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/AuthHeaderSigner.kt:57]: Could we add a comment here like we have for `DefaultAuthHeaderSigner` ?

File: wallet-sdk/src/test/kotlin/org/stellar/walletsdk/AuthTest.kt [hunk 2]
```diff
@@ -58,4 +68,88 @@ internal class AuthTest : SuspendTest() {
...
+
+  @Test
+  fun `create token test non custodial`() = runTest {
+    val accountKp =
+      SigningKeyPair.fromSecret("SC6UWRT6SMC7DSGJR67JTYHJQ6GPSGCKYJ66V5AVCLRGSGKXUNW5OXX7")
+    val domainKp =
+      SigningKeyPair.fromSecret("SCYVDFYEHNDNTB2UER2FCYSZAYQFAAZ6BDYXL3BWRQWNL327GZUXY7D7")
+    // Signing with a domain signer
+    val signer =
+      object : DefaultAuthHeaderSigner() {
```
[the remark this answers is not in the document]
Comment (Ifropc) [reply wallet-sdk/src/test/kotlin/org/stellar/walletsdk/AuthTest.kt:119]: Not really, DomainAuthSigner sends request to remote server, while this signer does signing with a specific key

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/AuthHeaderSigner.kt (hunk 1 above)
Comment (Ifropc) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/AuthHeaderSigner.kt:57]: Done

Comment (CassioMG) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/AuthHeaderSigner.kt:57]: Thanks! Have you pushed it? I don't see the change just yet

Comment (Ifropc) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/AuthHeaderSigner.kt:61]: Sorry didn't push 😅

**Review by lijamie98 — APPROVED** [review]
LGTM

**Pull request merged**